kornia/augmentation/functional/functional.py

```diff
@@ -44,13 +51,8 @@ def random_hflip(input: torch.Tensor, p: float = 0.5, return_transform: bool = F
     See :func:`~kornia.augmentation.random_generator.random_prob_generator` for details.
     See :func:`~kornia.augmentation.functional.apply_hflip` for details.
     """
-    input = _transform_input(input)
-    batch_size, _, h, w = input.size()
-    params = rg.random_prob_generator(batch_size, p=p)
-    output = apply_hflip(input, params)
-    if return_transform:
-        return output, compute_hflip_transformation(input, params)
-    return output
+    raise NotImplementedError(
+        f"functional random augmentation is deprecated. Please use `RandomHorizontalFlip`")
```

Comment:
I'm fine simplifying the API, but shouldn't we first add the deprecation warning and later remove ? What is the policy to remove functionals ? because I've seen you don't remove all of them.

Reply:
I only removed those ```random_*``` stuff. Throw a deprecation warning is not easy here since they are hard to maintain if we put some logic inside the ```AugmentationBase``` modules.

I am still working on these whole things that I haven't decided regarding the best practice here.